src/server/pkg/ppsutil/util.go

```diff
@@ -146,6 +147,16 @@ func GetPipelineInfo(pachClient *client.APIClient, name string, ptr *pps.EtcdPip
 	return result, nil
 }
 
+// GetPipelineInfo retrieves and returns a valid PipelineInfo from PFS. It does
+// the PFS read/unmarshalling of bytes as well as filling in missing fields
+func GetPipelineInfo(pachClient *client.APIClient, name string, ptr *pps.EtcdPipelineInfo) (*pps.PipelineInfo, error) {
```

Comment:
Why not just add a new bool field `allowIncomplete` to this?

Reply:
Because a bool field isn't as easy to parse as something unsafe.  If go had named parameters that would be great, but this makes it obvious if the function will return something unexpected.

Reply:
Makes sense!